conductor/lib/maya_utils.py

```diff
@@ -1038,13 +992,243 @@ def _scrape_ass_file(ass_filepath, node_attrs):
+
+            if node_type == "xgen_procedural":
+                logger.debug("Parsing xgen_procedural..")
+                xgen_paths = scrape_arnold_xgen_data_str(value)
+                paths.extend(xgen_paths)
+
+            elif value:
                 paths.append(value)
 
     arnold.AiNodeIteratorDestroy(iterator)
     return paths
 
 
+def scrape_arnold_xgen_data_str(string):
+    '''
+    Parse the given xgen_procedural.data string value into separate parts, and scrape each part
+    for dependencies
+    '''
+    paths = set()
+
+    # First convert the string of arg
+    xgen_args = parse_xgen_command(string)
+    for flag in ("file", 'geom'):
+        logger.debug('Reading "%s" flag', flag)
+        if flag not in xgen_args:
+            logger.warning('"%s" flag not found in xgen procedural.  Skipping', flag)
+            continue
+        path = xgen_args[flag]
+        logger.debug("path: %s", path)
+        if not path:
+            logger.warning('"%s" does not have a path specied', flag)
+            continue
+        paths.add(path)
+
+    # Parse xgen file dependencies
+    xgen_dependencies = set()
+    for path in paths:
+        if path.lower().endswith(".xgen"):
+            logger.debug("Parsing xgen dependencies for: %s", path)
+            if not os.path.isfile(path):
+                logger.warning("File does not exist, skipping: %s", path)
+                continue
+            xgen_dependencies.update(scrape_xgen_file(path))
+
+    return list(paths | xgen_dependencies)
+
+
+def parse_xgen_command(cmd_str):
+    '''
+    Parse xgen produceral command/args. Very crude with many assumptions.
+
+    args:
+        cmd_str: str. The string to parse, e.g. 
+        '-debug 1 -warning 1 -stats 1  -frame 1.000000  -nameSpace cat_main -file /tmp/cat.xgen -palette cat_xgen_coll -geom /tmp/cat_xgen_coll.abc -patch Paw_R_emitter -description catcuff_xgen_desc -fps 24.000000  -motionSamplesLookup -0.250000 0.250000  -motionSamplesPlacement -0.250000 0.250000  -world 1;0;0;0;0;1;0;0;0;0;1;0;0;0;0;1'
+
+    return: dict:, e.g.
+        {'debug': '1',
+         'description': 'catcuff_xgen_desc',
+         'file': '/tmp/cat.xgen',
+         'fps': '24.000000',
+         'frame': '1.000000',
+         'geom': '/tmp/cat_xgen_coll.abc',
+         'motionSamplesLookup': ['-0.250000', '0.250000'],
+         'motionSamplesPlacement': ['-0.250000', '0.250000'],
+         'nameSpace': 'cat_main',
+         'palette': 'cat_xgen_coll',
+         'patch': 'Paw_R_emitter',
+         'stats': '1',
+         'warning': '1',
+         'world': '1;0;0;0;0;1;0;0;0;0;1;0;0;0;0;1'}
+
+    '''
+    args = collections.defaultdict(list)
+    attr = None
+    for part in shlex.split(cmd_str):
+        if part.startswith("-") and not _is_number(part):
+            attr = part.lstrip("-")
+        else:
+            args[attr].append(part)
+
+    command_args = {}
+    for flag, value in args.iteritems():
+        if len(value) is 1:
```

Comment:
`if len(value) == 1:` is more future-proof. For example, if someone changes this line to `if len(value) is some_value:`, then even if `some_value` has the value of `1`, it won't have the same `id` as `1` and the condition will be `False`.

Reply:
## tldr
the choice of using `is` was intentional, but is probably a bad idea.

## not all integers are created equally
The reason I used `is` was because my (erroneous) understanding was that _every_ integer in python essentially acts as singleton object, meaning there is only one object that represents the integer value of 1, for example.  
```python
one = 1
id(one) == id(1) # True
```
But my understanding of this behavior is not entirely true; it turns out that only integer values `-5` through  `256 ` behave in this singleton manner (oddly enough).  so if you use an integer that is outside of that range, you'll get a different behavior:
```python
twofiftyseven = 257
id(twofiftyseven) == id(257) # False
```
Very strange.  So that's a new lesson for me!


## equality is not always safe
But as to why I didn't simply use the `==` operator from the very start (as you suggested), I thought I _was_ being _future proof_!  Using the `==` operator simply calls the object's `__eq__` method.  And that can lead to misleading/unexpected results.  
Consider this unusual case:

```python
class EqualityLiar(object):
     def __eq__(self, other):
         # Always report that you're equal
         return True

other_value = EqualityLiar()

other_value == 1 # True    # lies!
other_value == 2 # True    # lies!
other_value is 1 # False   # truth!
other_value is 2 # False   # truth!
```

That being said, in light of the inconsistent manner that integer objects are created/behave (with respect to their ids), I don't believe this corner-case benefit of using `is` over  `==` is justified.  In other words, `==` for the win!  i'll make the change. Thanks for your suggestion!